src/main/java/core/db/DBConfigUpdater.java

```diff
@@ -61,8 +61,9 @@ private static void updateDBConfig5(double configVersion, boolean alreadyApplied
 			try {
 				String sql = "SELECT TRAININGDATE FROM XTRADATA ORDER BY TRAININGDATE ASC LIMIT 1";
 				ResultSet rs = m_clJDBCAdapter.executeQuery(sql);
+				assert rs != null;
```

Comment:
I don't know if this does what you intend.  By default assertions are not enabled.  When they are enabled, this will throw an error if the condition is not met; it would be more appropriate to handle the `null` case.

Cf. https://docs.oracle.com/javase/specs/jls/se17/html/jls-14.html#jls-14.10

Reply:
@tychobrailleur tbh, this was a suggestion of intellijs review method. i think it is a kind of TODO: Check null pointer handling here.

Thought it would allow for a slightly faster problem finding when debugging